supportedBackends/google/pipelines/common/src/main/scala/cromwell/backend/google/pipelines/common/PipelinesApiBackendLifecycleActorFactory.scala

```diff
@@ -82,4 +90,48 @@ abstract class PipelinesApiBackendLifecycleActorFactory(override val name: Strin
 object PipelinesApiBackendLifecycleActorFactory {
   val preemptionCountKey = "PreemptionCount"
   val unexpectedRetryCountKey = "UnexpectedRetryCount"
+
+  private val logger: Logger = LoggerFactory.getLogger(getClass)
+
+  private [common] def robustBuildAttributes(buildAttributes: () => PipelinesApiConfigurationAttributes,
+                                             maxAttempts: Int = 3,
+                                             initialIntervalMillis: Int = 5000,
+                                             maxIntervalMillis: Int = 10000,
+                                             multiplier: Double = 1.5,
+                                             randomizationFactor: Double = 0.5): PipelinesApiConfigurationAttributes = {
+    val backoff = new ExponentialBackOff.Builder()
+      .setInitialIntervalMillis(initialIntervalMillis)
+      .setMaxIntervalMillis(maxIntervalMillis)
+      .setMultiplier(multiplier)
+      .setRandomizationFactor(randomizationFactor)
+      .build()
+
+    // Is this an `Exception` (as opposed to an `Error`) with a message indicating the operation should be retried?
+    def isRetryableException(t: Throwable): Boolean = {
+      t match {
+        case e: Exception =>
+          Option(e.getMessage) match {
+            case Some(message) => message.contains("We encountered an internal error. Please try again.")
+            case None => false
+          }
+        case _ => false
+      }
+    }
+
+    // `attempt` is 1-based
+    @tailrec
+    def build(attempt: Int): PipelinesApiConfigurationAttributes = {
+      Try {
+        buildAttributes()
+      } match {
+        case Success(value) => value
+        case Failure(t) if isRetryableException(t) && attempt < maxAttempts =>
```

Comment:
Is all this `case`/`match`ing functionally equivalent to a `recoverWith`?

Reply:
ish... I can do this as a `recoverWith` but despite looking tail recursive it loses the `@tailrec`. Not really a big deal since we're recursing at most a few times.